### ISSUE TITLE
Not compiles wasm target on Windows

### ISSUE THREAD

**Issue by Fedoresko** [gyscos/zstd-rs#151]
````
error: failed to run custom build command for 'zstd-sys v2.0.1+zstd.1.5.2'

Caused by:
  process didn't exit successfully: 'F:\work\web-cycling\target\release\build\zstd-sys-a05007c57adc943b\build-script-build' (exit code: 1)
  --- stdout
  cargo:rustc-cfg=feature="std"
  cargo:rerun-if-changed=wasm-shim/stdlib.h
  cargo:rerun-if-changed=wasm-shim/string.h
  TARGET = Some("wasm32-unknown-unknown")
  HOST = Some("x86_64-pc-windows-msvc")
  CC_wasm32-unknown-unknown = None
  CC_wasm32_unknown_unknown = None
  TARGET_CC = None
  CC = None
  CFLAGS_wasm32-unknown-unknown = None
  CFLAGS_wasm32_unknown_unknown = None
  TARGET_CFLAGS = None
  CFLAGS = None
  CRATE_CC_NO_DEFAULTS = None
  DEBUG = Some("false")
  running: "**clang**" "-O3" "-ffunction-sections" "-fdata-sec.....
````

should use cl not clang for this host.

**Comment by gyscos** [comment]
The compiler is chosen by the [`cc-rs`](https://github.com/rust-lang/cc-rs/) crate.

May be related to https://github.com/rust-lang/cc-rs/issues/681
TLDR: You might need to install `clang` and `ar`.

**Comment by REASY** [comment]
@Fedoresko if you use Visual Studio Tools (2022) with cmake component installed, you can run `Developer Command Prompt for VS 2022`  you just need to set `ar=llvm-ar` and it will work:
![image](https://user-images.githubusercontent.com/5107562/224533926-3e41510c-1aef-4831-9137-2565e61c5b95.png)

**Comment by REASY** [comment]
@gyscos is it expected that wasmtime cannot run the example [zstd-safe/zstd-sys/examples/it_work.rs](https://github.com/gyscos/zstd-rs/blob/83d21974bf1d91431f63f57a82ad6ff28dd57e2d/zstd-safe/zstd-sys/examples/it_work.rs) on commit 83d21974bf1d91431f63f57a82ad6ff28dd57e2d 
```
root@DESKTOP-FONP7QD:/mnt/c/repos/github/gyscos/zstd-rs/zstd-safe/zstd-sys# git log -n 1
commit 83d21974bf1d91431f63f57a82ad6ff28dd57e2d (HEAD -> main, origin/main, origin/HEAD)
Author: Chris Burgess <9002722+cgbur@users.noreply.github.com>
Date:   Sat Feb 25 15:30:11 2023 -0500

    Fix OutBuf around_pos invariant (#215)
root@DESKTOP-FONP7QD:/mnt/c/repos/github/gyscos/zstd-rs/zstd-safe/zstd-sys# clang --version
clang version 10.0.0-4ubuntu1
Target: x86_64-pc-linux-gnu
Thread model: posix
InstalledDir: /usr/bin
root@DESKTOP-FONP7QD:/mnt/c/repos/github/gyscos/zstd-rs/zstd-safe/zstd-sys# wasmtime --version
wasmtime-cli 6.0.1
root@DESKTOP-FONP7QD:/mnt/c/repos/github/gyscos/zstd-rs/zstd-safe/zstd-sys# cargo build --target wasm32-unknown-unknown --no-default-features --example it_work
    Finished dev [unoptimized + debuginfo] target(s) in 0.03s
root@DESKTOP-FONP7QD:/mnt/c/repos/github/gyscos/zstd-rs/zstd-safe/zstd-sys# ls -la target/wasm32-unknown-unknown/debug/examples/
total 19328
drwxrwxrwx 1 root root    4096 Mar 12 16:44 .
drwxrwxrwx 1 root root    4096 Mar 12 11:50 ..
-rwxrwxrwx 1 root root     390 Mar 12 16:27 it_work-0e1a13e5f0f45118.d
-rwxrwxrwx 2 root root 6606331 Mar 12 16:27 it_work-0e1a13e5f0f45118.wasm
-rwxrwxrwx 1 root root     382 Mar 12 16:40 it_work-baeeb6a126d4dc5b.d
-rwxrwxrwx 1 root root 6577359 Mar 12 16:40 it_work-baeeb6a126d4dc5b.wasm
-rwxrwxrwx 1 root root     676 Mar 12 16:44 it_work.d
-rwxrwxrwx 2 root root 6606331 Mar 12 16:27 it_work.wasm
root@DESKTOP-FONP7QD:/mnt/c/repos/github/gyscos/zstd-rs/zstd-safe/zstd-sys# wasmtime target/wasm32-unknown-unknown/debug/examples/it_work.wasm  --invoke test_compress
Error: failed to run main module `target/wasm32-unknown-unknown/debug/examples/it_work.wasm`

Caused by:
    0: failed to instantiate "target/wasm32-unknown-unknown/debug/examples/it_work.wasm"
    1: unknown import: `env::malloc` has not been defined
root@DESKTOP-FONP7QD:/mnt/c/repos/github/gyscos/zstd-rs/zstd-safe/zstd-sys#
```
![image](https://user-images.githubusercontent.com/5107562/224534104-df879ecd-8df9-42ce-8f36-8d9c5ec73110.png)

**Comment by REASY** [comment]
I went back to git history and found that works 417d4420e3e7a8c406b8641e731caa427d98398d
![image](https://user-images.githubusercontent.com/5107562/224536710-27160be8-a12c-406b-a362-1588cb1f8b64.png)

**Comment by jhult** [comment]
Related: #93
